BaragonData/src/main/java/com/hubspot/baragon/data/AbstractDataStore.java

```diff
@@ -152,14 +153,14 @@ public T apply(byte[] data) {
     }
   }
 
-  protected <T> T deserialize(byte[] data, Class<T> klass) {
+  protected <T> T deserialize(byte[] data, Class<T> klass, String path) {
     try {
       return objectMapper.readValue(data, klass);
     } catch (JsonParseException jpe) {
       try {
-        LOG.error(String.format("Invalid Json: %s", objectMapper.readValue(data, String.class)));
+        LOG.error("Invalid Json at path {}: {}", path, new String(data, StandardCharsets.UTF_8), jpe);
       } catch (Exception e) {
-        LOG.error("Could not get raw json string", e);
+        LOG.error("Could not parse valid json at path {}", path, e);
```

Comment:
Ah, gotcha. We should probably update the error message -- "parse valid json" seems a little misleading. LGTM other than that.